generated/nirfsg/nirfsg/session.py

```diff
@@ -52,43 +52,1095 @@ def __exit__(self, exc_type, exc_value, traceback):
+
+    Example: :py:attr:`my_session.script_trigger_type`
+    '''
+    def __init__(self, session, repeated_capability_list):
+        object.__setattr__(self, '_session', session)
+        object.__setattr__(self, '_repeated_capability_list', repeated_capability_list)
+        object.__setattr__(self, '_prefix', 'scripttrigger')
+        object.__setattr__(self, '_current_repeated_capability_list', repeated_capability_list if len(repeated_capability_list) > 0 else [''])
+        object.__setattr__(self, '_separator', '')
+
+    def __setattr__(self, key, value):
+        if key not in dir(self):
+            raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, key))
+        object.__setattr__(self, key, value)
+
+    def __getitem__(self, repeated_capability):
+        '''Set/get properties or call methods with a repeated capability (i.e. channels)'''
+        rep_caps_list = _converters.convert_repeated_capabilities(repeated_capability, self._prefix)
+        complete_rep_cap_list = [
+            current_rep_cap + self._separator + rep_cap
+            for current_rep_cap in self._current_repeated_capability_list
+            for rep_cap in rep_caps_list
+        ]
+        object.__setattr__(self, '_current_repeated_capability_list', complete_rep_cap_list)
+        self._current_repeated_capability_list = complete_rep_cap_list
+
+        return self
+
+    def _get_attribute_vi_real64(self, attribute):
+        repeated_capability = ','.join(self._current_repeated_capability_list)
+        value = self._session._interpreter.get_attribute_vi_real64(repeated_capability, attribute)
+        return value
+
+    def _set_attribute_vi_real64(self, attribute, value):
+        repeated_capability = ','.join(self._current_repeated_capability_list)
+        self._session._interpreter.set_attribute_vi_real64(repeated_capability, attribute, value)
+
+    def _get_attribute_vi_int32(self, attribute):
+        repeated_capability = ','.join(self._current_repeated_capability_list)
+        value = self._session._interpreter.get_attribute_vi_int32(repeated_capability, attribute)
+        return value
+
+    def _set_attribute_vi_int32(self, attribute, value):
+        repeated_capability = ','.join(self._current_repeated_capability_list)
+        self._session._interpreter.set_attribute_vi_int32(repeated_capability, attribute, value)
+
+    def _get_attribute_vi_string(self, attribute):
+        repeated_capability = ','.join(self._current_repeated_capability_list)
+        value = self._session._interpreter.get_attribute_vi_string(repeated_capability, attribute)
+        return value
+
+    def _set_attribute_vi_string(self, attribute, value):
+        repeated_capability = ','.join(self._current_repeated_capability_list)
+        self._session._interpreter.set_attribute_vi_string(repeated_capability, attribute, value)
```

Comment:
If you're going to put these in every rep cap class, why not have a base class to inherit them from?